mteb/evaluation/evaluators/RetrievalEvaluator.py

```diff
@@ -1,257 +1,240 @@
-        corpus: Dict[str, str],  # cid => doc
-        relevant_docs: Dict[str, Set[str]],  # qid => Set[cid]
-        corpus_chunk_size: int = 50000,
-        mrr_at_k: List[int] = [10],
-        ndcg_at_k: List[int] = [10],
-        accuracy_at_k: List[int] = [1, 3, 5, 10],
-        precision_recall_at_k: List[int] = [1, 3, 5, 10],
-        map_at_k: List[int] = [100],
-        show_progress_bar: bool = False,
-        batch_size: int = 32,
-        name: str = "",
-        score_functions: List[Callable[[torch.Tensor, torch.Tensor], torch.Tensor]] = {
-            "cos_sim": cos_sim,
-            "dot": dot_score,
-        },  # Score function, higher=more similar
-        main_score_function: str = None,
-        limit: int = None,
-        **kwargs
-    ):
+            self, retriever = None, k_values: List[int] = [1,3,5,10,100,1000], score_function: str = "cos_sim", **kwargs
+        ):
         super().__init__(**kwargs)
-        self.queries_ids = []
-        for qid in queries:
-            if qid in relevant_docs and len(relevant_docs[qid]) > 0:
-                self.queries_ids.append(qid)
-                if limit and len(self.queries_ids) >= limit:
-                    break
-
-        self.queries = [queries[qid] for qid in self.queries_ids]
-
-        self.corpus_ids = list(corpus.keys())
-        self.corpus = [corpus[cid] for cid in self.corpus_ids]
-
-        self.relevant_docs = relevant_docs
-        self.corpus_chunk_size = corpus_chunk_size
-        self.mrr_at_k = mrr_at_k
-        self.ndcg_at_k = ndcg_at_k
-        self.accuracy_at_k = accuracy_at_k
-        self.precision_recall_at_k = precision_recall_at_k
-        self.map_at_k = map_at_k
-
-        self.show_progress_bar = show_progress_bar
-        self.batch_size = batch_size
-        self.name = name
-        self.score_functions = score_functions
-        self.score_function_names = sorted(list(self.score_functions.keys()))
-        self.main_score_function = main_score_function
-
-    def __call__(self, model) -> float:
-        scores = self.compute_metrics(model)
-
-        if self.main_score_function is None:
-            scores["main_score"] = max(
-                [scores[name]["map_at_" + str(max(self.map_at_k))] for name in self.score_function_names]
-            )
-        else:
-            scores["main_score"] = scores[self.main_score_function]["map_at_" + str(max(self.map_at_k))]
-        return scores
-
-    def compute_metrics(self, model, corpus_model=None, corpus_embeddings: torch.Tensor = None) -> Dict[str, float]:
-        if corpus_model is None:
-            corpus_model = model
-
-        max_k = max(
-            max(self.mrr_at_k),
-            max(self.ndcg_at_k),
-            max(self.accuracy_at_k),
-            max(self.precision_recall_at_k),
-            max(self.map_at_k),
+        self.retriever = DenseRetrievalExactSearch(
+            retriever if is_dres_compatible(retriever) else DRESModel(retriever),
```

Comment:
Should we log something when DRESModel is used as default ? Like "Queries and corpus will be encoded via the same method"